graphql_client_codegen/src/lib.rs

```diff
@@ -144,19 +151,57 @@ pub fn generate_module_token_stream(
     };
 
     let module_name = Ident::new(
-        options.struct_name.to_snake_case().as_str(),
+        options
+            .operation_name
+            .clone()
+            .unwrap_or_else(|| options.module_name.clone().unwrap())
+            .to_snake_case()
+            .as_str(),
         Span::call_site(),
     );
-    let struct_name = Ident::new(options.struct_name.as_str(), Span::call_site());
-    let schema_output = codegen::response_for_query(
-        schema,
-        query,
-        &operation,
-        response_derives,
-        deprecation_strategy,
-    )?;
-
-    let result = quote!(
+
+    let mut schema_and_operations = Vec::with_capacity(operations.len());
+
+    for operation in operations.iter() {
+        let schema_output = codegen::response_for_query(
+            schema.clone(),
+            query.clone(),
+            &operation,
+            response_derives.clone(),
+            deprecation_strategy.clone(),
+        )?;
+        schema_and_operations.push((schema_output, operation.name.as_str()));
+    }
+
+    let result = build_module_token_stream(
+        &module_name.to_string(),
+        &query_string,
+        schema_and_operations,
+    );
+
+    Ok(result)
+}
+
+fn build_module_token_stream(
+    module_name: &str,
+    query_string: &str,
+    schema_and_operations: Vec<(TokenStream, &str)>,
+) -> TokenStream {
+    let common_token_stream = &build_common_token_stream(module_name, query_string).to_string();
+    let mut query_struct_token_streams = vec![];
+    for (schema_output, operation_name) in schema_and_operations {
+        let token_stream =
+            build_query_struct_token_stream(module_name, schema_output, operation_name);
+        query_struct_token_streams.push(token_stream.to_string());
+    }
+    quote!(
+        #common_token_stream
+        #(#query_struct_token_streams)*
+    )
```

Comment:
@tomhoule 
It did not work as it would be string.
I really want to use [concat](https://manishearth.github.io/rust-internals-docs/syntax/tokenstream/struct.TokenStream.html), but we can only use it nightly.